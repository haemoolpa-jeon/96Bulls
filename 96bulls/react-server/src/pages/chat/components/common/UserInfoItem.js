import React, { Component } from 'react';
import styles from './UserInfoItem.module.scss';
import PropTypes from 'prop-types';

class UserInfoItem extends Component {
  render() {
    const { user, onClick } = this.props;
    return (
      <button className={styles['user-info-item']} onClick={onClick} type="button">
        <img className={styles.avatar} src={user.avatar} alt={`${user.name} avatar`} />
        <div className={styles['user-info-content']}>
          <span className={styles['display-name']}>{user.name}</span>
          <span className={styles.email}>{user.email}</span>
        </div>
      </button>
    );
  }
}

UserInfoItem.propTypes = {
  user: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default UserInfoItem;
