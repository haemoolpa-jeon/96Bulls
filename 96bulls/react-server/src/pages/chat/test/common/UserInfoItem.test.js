import React from 'react';
import UserInfoItem from '../../components/common/UserInfoItem';
import {shallow} from 'enzyme/build';

describe('UserInfoItem', () => {
  let component = null;
  const user = {
    avatar: 'http://wwww.mock-image-url.com',
    name: 'myp',
    email: 'myp7@daum.net'
  };

  const mockCallback = jest.fn(e => e);
  it('초기 렌더링이 문제없이 되야함', () => {
    component = shallow(<UserInfoItem
      user={user}
      onClick={mockCallback}/>);
  });

  it('초기 렌더링 스냅샷 일치함', () => {
    expect(component).toMatchSnapshot();
  });

  it('onClick 함수가 수행되어야 함', () => {
    component.find('button').simulate('click');
    expect(mockCallback.mock.calls.length).toBe(1);
  });
});
