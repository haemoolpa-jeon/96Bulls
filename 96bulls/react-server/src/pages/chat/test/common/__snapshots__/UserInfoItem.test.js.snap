// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoItem 초기 렌더링 스냅샷 일치함 1`] = `
<button
  className="user-info-item"
  onClick={[MockFunction]}
  type="button"
>
  <img
    alt="myp avatar"
    className="avatar"
    src="http://wwww.mock-image-url.com"
  />
  <div
    className="user-info-content"
  >
    <span
      className="display-name"
    >
      myp
    </span>
    <span
      className="email"
    >
      myp7@daum.net
    </span>
  </div>
</button>
`;
